src/js/elements/mapchip/mapchip.js

```diff
@@ -75,14 +101,87 @@ export class MapChip extends Component {
         this._titleArea = $(this.container).find('.filters__header_name div');
         this._filtersContainer = $(this.container).find(filterContentClass);
         this._legendContainer = $(this.container).find(legendContainerClass);
+        this._filterHeaderTitleContainer = $(this.container).find(filterHeaderClass);
+        this._filterHeaderToggleContainer = $(this.container).find(filterHeaderToggleClass);
+    }
+
+    getFilterLabelHTML (){
+      let html = "<div class='filters__header_label' title='Show Applied Filters'>";
+      html += "<span id='selected_filter_count'>0</span>"
+      html += " of "
+      html += "<span id='total_filter_count'></span>"
+      html += " filters applied"
+      html += "</div>"
+
+      return html;
+    }
+
+    updateDomElements() {
+        const filterLabelHTML = this.getFilterLabelHTML();
+        this._filterHeaderTitleContainer.find(".filters__header_name").css("width", "auto");
+        this._filterHeaderTitleContainer.append(filterLabelHTML);
+        this._filterHeaderLabelContainer = this._filterHeaderTitleContainer.find(".filters__header_label");
+        this._selectedFilterCountContainer = this._filterHeaderLabelContainer.find("#selected_filter_count");
+        this._totalFilterCountContainer = this._filterHeaderLabelContainer.find("#total_filter_count");
+        this._toggleIconDownContainer = this._filterHeaderToggleContainer.find(".toggle-icon-v--first");
+        this._toggleIconUpContainer = this._filterHeaderToggleContainer.find(".toggle-icon-v--last");
+
+        // Update right side menu
+        this.updateToggleIcon();
+        this.addInfoIcon();
+
+        this._descriptionInfoIconContainer = $(this._container).find(".filters__header_info");
+        this._descriptionInfoIconContainer.tooltip({
+          position: {
+            my: "center bottom-20",
+            at: "center top",
+            using: function( position, feedback ) {
+              $( this ).css( position );
+              $( "<div>" )
+                .addClass( "arrow" )
+                .addClass( feedback.vertical )
+                .addClass( feedback.horizontal )
+                .appendTo( this );
+            }
+          }
+        });
+        this._filterHeaderLabelContainer.tooltip({
+          position: {
+            my: "center bottom-20",
+            at: "center top",
+            using: function( position, feedback ) {
+              $( this ).css( position );
+              $( "<div>" )
+                .addClass( "arrow" )
+                .addClass( feedback.vertical )
+                .addClass( feedback.horizontal )
+                .appendTo( this );
+            }
+          }
```

Comment:
Another example of where a library should handle components, even if it's just some custom library like mapchip.components.js

Reply:
hmm... I think it's a good idea @michaelglenister 
but tooltip should be a global component so should i create a component folder in src and add tooltip component there ?

We will pass the element and position to it.
@emre2038 Do you think it's a good step forward regarding code sanitization and segregation?

Reply:
I liked that you separated tooltip into a new file. we have `js > ui_components` folder for the components that will be used from multiple widgets. does it make sense to move the new components you've created into that folder?

also while naming the files we use snack_case

Reply:
Other components are very specific to mapchip. So i added a folder there only
We can have component folder when we get a component library and move these properly to parent folder
Also will change filename @emre2038 👍

Reply:
cool 👍